boefjes/boefjes/plugins/kat_shodan/normalize.py

```diff
@@ -12,21 +13,26 @@ def run(normalizer_meta: NormalizerMeta, raw: Union[bytes, str]) -> Iterator[OOI
     results = json.loads(raw)
     ooi = Reference.from_str(normalizer_meta.raw_data.boefje_meta.input_ooi)
 
-    for scan in results["data"]:
-        port_nr = scan["port"]
-        transport = scan["transport"]
+    if not results:
+        logging.info("No Shodan results available for normalization.")
+    elif "data" not in results:
+        raise ValueError("Shodan results exist without data.")
```

Comment:
Why should the normalizer fail? No result is usually also a good execution.

Doing this also means there is no easy way to differentiate (in the task list) between a failure due to programming/input errors or missing results

Reply:
Exactly! If there are no results there is a good execution and only a logging message is added. However, if there are results but the dictionary key `data` is missing; that would be unexpected and would justify a visual issue in the front-end in my opinion.